File: 3704-count-partitions-with-even-sum-difference/3704-count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total_sum = sum(nums)
        cnt = 0
        prefix_sum = 0
        
        for i in range(len(nums) - 1):  # Exclude the last partition
            prefix_sum += nums[i]
            suffix_sum = total_sum - prefix_sum
            
            if abs(prefix_sum - suffix_sum) % 2 == 0:
                cnt+=1
        return cnt

File: 3704-count-partitions-with-even-sum-difference/3704-count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int sm = 0;
        for(int num:nums){
            sm += num;
        }
        int n = nums.length;
        if(sm % 2 == 0){
            return n -1;
        }
        return 0;
    }
}